### PR TITLE
Return pattern sorts from verifier

### PR DESCRIPTION
**Pull request by ttuegel**
The definition verifier now returns patterns annotated with their proper sorts. The sort annotations are not yet used, but will eventually replace `symbolOrAliasSorts` from `MetadataTools`.

###### Reviewer checklist

- [ ] Test coverage: `stack test --coverage`
- [ ] Public API documentation: `stack haddock`
- [ ] Style conformance: `stylish-haskell`

---

### CODE REVIEW

File: src/main/haskell/kore/src/Kore/Parser/ParserImpl.hs [hunk 1]
```diff
@@ -401,6 +401,25 @@ symbolOrAliasPatternRemainderParser childParser x identifier =
         <*> inParenthesesListParser childParser
         )
 
+symbolOrAliasParser
```
Comment (virgil-serbanuta) [inline src/main/haskell/kore/src/Kore/Parser/ParserImpl.hs]: Why not `headParser`?

File: src/main/haskell/kore/test/Test/Kore/MetaML/Lift.hs [hunk 2]
```diff
@@ -452,87 +452,6 @@ test_lift =
                 }
             )
         )
-    , testCase "Lift Meta Alias Declaration"
```
Comment (virgil-serbanuta) [inline src/main/haskell/kore/test/Test/Kore/MetaML/Lift.hs]: Why did you delete these tests?

File: src/main/haskell/kore/test/Test/Kore/Unparser.hs [hunk 3]
```diff
@@ -32,30 +32,6 @@ test_unparse =
                 )
             )
             "sort x{} []"
-        , unparseTest
```
Comment (virgil-serbanuta) [inline src/main/haskell/kore/test/Test/Kore/Unparser.hs]: Why remove this test?

File: src/main/haskell/kore/src/Kore/Parser/ParserImpl.hs (hunk 1 above)
Comment (ttuegel) [reply src/main/haskell/kore/src/Kore/Parser/ParserImpl.hs]: I used "symbolOrAlias" because that is the name of the type, but I'm not attached to the name.

File: src/main/haskell/kore/test/Test/Kore/MetaML/Lift.hs (hunk 2 above)
Comment (ttuegel) [reply src/main/haskell/kore/test/Test/Kore/MetaML/Lift.hs]: The alias declarations in these tests are not syntactically-valid Kore (650ce2a32f90f975cb3a5aab787de91187c33876).

File: src/main/haskell/kore/test/Test/Kore/Unparser.hs (hunk 3 above)
Comment (ttuegel) [reply src/main/haskell/kore/test/Test/Kore/Unparser.hs]: The sentence that this test unparses is not valid Kore (650ce2a32f90f975cb3a5aab787de91187c33876).

File: src/main/haskell/kore/src/Kore/AST/Error.hs [hunk 4]
```diff
@@ -45,8 +44,12 @@ koreFailWithLocations locations errorMessage =
 containing the provided locations whenever the provided flag is true.
 -}
 koreFailWithLocationsWhen
-    :: AstWithLocation astWithLocation
-    => Bool -> [astWithLocation] -> String -> Either (Error a) ()
+    :: MonadError (Error e) m
+    => AstWithLocation astWithLocation
```
[the remark this answers is not in the document]
Comment (ttuegel) [reply src/main/haskell/kore/src/Kore/AST/Error.hs]: That would be a typo! :laughing:

File: src/main/haskell/kore/src/Kore/Parser/ParserImpl.hs (hunk 1 above)
Comment (virgil-serbanuta) [reply src/main/haskell/kore/src/Kore/Parser/ParserImpl.hs]: Sorry, I meant: Why not the existing `headParser` function?
